src/NaCl.Core/Base/Salsa20Base.cs

```diff
@@ -2,22 +2,37 @@
 {
     using System;
     using System.Runtime.CompilerServices;
-    using System.Security.Cryptography;
-
     using Internal;
+    using NaCl.Core.Base.SalsaCore;
 
     /// <summary>
     /// Base class for <seealso cref="NaCl.Core.Salsa20" /> and <seealso cref="NaCl.Core.XSalsa20" />.
     /// </summary>
     /// <seealso cref="NaCl.Core.Base.Snuffle" />
     public abstract class Salsa20Base : Snuffle
     {
+        readonly ISalsa20Core _salsa20Core;
+
         /// <summary>
         /// Initializes a new instance of the <see cref="Salsa20Base"/> class.
         /// </summary>
         /// <param name="key">The key.</param>
         /// <param name="initialCounter">The initial counter.</param>
-        protected Salsa20Base(ReadOnlyMemory<byte> key, int initialCounter) : base(key, initialCounter) { }
+        protected Salsa20Base(ReadOnlyMemory<byte> key, int initialCounter) : base(key, initialCounter)
+        {
+#if INTRINSICS

```

Comment:
I wonder if we do need the preprocessor directive since we are checking if the hardware supports it